05-onchain.md

```diff
@@ -587,16 +587,37 @@ Note: even if the `to_remote` output is not swept, the resulting
 
 # Generation of HTLC Transactions
 
-If `option_anchor_outputs` does not apply to the commitment transaction, then HTLC-timeout and HTLC-success transactions are complete transactions with (hopefully!) reasonable fees and must be used directly.
+If `option_anchors` does not apply to the commitment transaction, then
+HTLC-timeout and HTLC-success transactions are complete transactions with
+(hopefully!) reasonable fees and must be used directly.
 
-Otherwise, the use of `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY` on the HTLC signatures received from the peer allows HTLC transactions to be combined with other transactions.
-The local signature MUST use `SIGHASH_ALL`, otherwise anyone can attach additional inputs and outputs to the tx.
+Otherwise, the use of `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY` on the HTLC
+signatures received from the peer allows HTLC transactions to be combined with
+other transactions.  The local signature MUST use `SIGHASH_ALL`, otherwise
+anyone can attach additional inputs and outputs to the tx.
+
+If `option_anchors_zero_fee_htlc_tx` applies, then the HTLC-timeout and
+HTLC-success transactions are signed with the input and output having the same
+value. This means they have a zero fee and MUST be combined with other inputs
+to arrive at a reasonable fee.
 
 ## Requirements
 
-A node which broadcasts an HTLC-success or HTLC-timeout transaction for a commitment transaction for which `option_anchor_outputs` applies:
-  - MUST contribute sufficient fee to ensure timely inclusion in a block.
-  - MAY combine it with other transactions.
+A node which broadcasts an HTLC-success or HTLC-timeout transaction for a
+commitment transaction:
+  1. if `option_anchor_outputs` applies:
+    - SHOULD combine it with inputs contributing sufficient fee to ensure
+      timely inclusion in a block.
+    - MAY combine it with other transactions.
+  2. if `option_anchors_zero_fee_htlc_tx` applies:
+    - MUST combine it with inputs contributing sufficient fee to ensure timely
+      inclusion in a block.
+    - MAY combine it with other transactions.
```

Comment:
This MAY is really unsafe, if the combined-with transaction includes an input from a third-party transaction you might have feerate manipulation (like big witness data). Or same if it pays to a third-party (pinning the first-broadcast HTLC-txn and blocking any replacement)